### DISCUSSION TITLE
Project init CLI

### DISCUSSION THREAD

**Discussion by tailhook** [geldata/gel#1789] [Ideas]
I propose to add a command:
```
edgedb project-init
```
That reads additional config file in our normal project structure:
```toml
# dbschema/project.toml
[dev]
instance-name = "myproj"
edgedb-version = "1.0"
```
Does the following:
1. Installs specified version of edgedb
2. Create an instance named `myproj`
3. Runs `edgedb -Imyproj migrate` to initialize the schema on that instance

Then we can have additional features:
1. Use the file in `curl https://sh.edgedb.com | sh` pipeline: Ask to initialize that instance if it's run from the project directory
2. Add `database_dump="./dev_data.dump"` that is restored before `edgedb migrate`. So that user has both prepopulated data and up to date schema.
3. Ask if user wants to use cloud or install edgedb locally

Note: this proposal has a nice side-effect that edgedb-version used is not mentioned at a random place in readme (if at all) and that we can search for such configs on github to find out what version at least part of ours users use.

**Comment by elprans** [comment]
I like the idea in general.  Would `project-init` create the initial structure in an empty project, or should there be two commands?

**Reply by tailhook** [reply]
I think single command is okay when the project is run by itself. But I'm not sure that's good in `curl | sh` pipeline, since that can be run in arbitrary place. We may try to detect if it's within a project probably by checking `.git` directory or similar.
